Update compare-static-analysis-results for better readability and EWS support

briannafan opened the pull request:
#### bff30458ca217d2d653f52f8ee29fda881c08337
<pre>
Update compare-static-analysis-results for better readability and EWS support
<a href="https://bugs.webkit.org/show_bug.cgi?id=276783">https://bugs.webkit.org/show_bug.cgi?id=276783</a>
<a href="https://rdar.apple.com/132017828">rdar://132017828</a>

Reviewed by Ryosuke Niwa.

Generate an unexpected_results.json when comparing results by run.
Update logging to use clearer wording and less unnecessary output.

* Tools/Scripts/compare-static-analysis-results:
(compare_project_results_to_expectations):
(compare_project_results_by_run):
(main):

Canonical link: <a href="https://commits.webkit.org/281096@main">https://commits.webkit.org/281096@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb30de398f7ab418fd1cd069ce08d73ab93da808

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58740 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38068 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11230 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/62371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/9184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60869 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45705 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9384 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/62371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/9184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60771 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/35633 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50797 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/62371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/32371 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/8093 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/8188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/54325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/8371 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/64073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2653 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/8363 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/64073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2662 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50821 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/64073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2228 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8754 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/33898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/34982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/36067 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/34728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Tools/Scripts/compare-static-analysis-results:
```diff
@@ -136,22 +136,28 @@ def compare_project_results_to_expectations(args, new_path, project):
+        if unexpected:
+            print(results_summary)
+        else:
+            print('No unexpected results!')
```
Maybe we can do this in one line:
```python
print(results_summary if unexpected else 'No unexpected results')
```

rniwa commented on Tools/Scripts/compare-static-analysis-results:
```diff
@@ -160,31 +166,48 @@ def compare_project_results_by_run(args, archive_path, new_path, project):
+            'Files fixed': fixed_files,
+            'Unexpected issues': new_issues,
+            'Unexpected failing files:': new_files
+        }.items():
```
Hm... I'm not sure that this format is more readable.
A bunch of print statements looked simpler.

briannafan replied:
Oh I meant for the logging output. Before we had lines for each checker even though there were no unexpected results. Now they only print if there are results.
I agree with the additional complexity, but now we can iterate and check each list.

rniwa replied:
I think the alternative is this.
```python
    if fixed_issues or fixed_files or new_issues or new_files:
        print(f'    Issues fixed: {len(fixed_issues)}')
        print(f'    Files fixed: {len(fixed_files)}')
        print(f'    New issues: {len(new_issues)}')
        print(f'    New files with issues: {len(new_files)}\n')
    else
        print('No unexpected results')
```

briannafan replied:
Sure, I guess we can list all the counts as long as one is nonzero.

rniwa replied:
oh I see. We're only outputting a non-zero line. I guess we can keep this code as is in that case.

briannafan replied:
Lol I just made your changes, I think my implementation really only saves one line or two and it helps to have standardized output for each checker so you don't need to read a line to figure out which one it is (if people will be looking at this at all).

briannafan committed:
Update compare-static-analysis-results for better readability and EWS support
https://bugs.webkit.org/show_bug.cgi?id=276783
rdar://132017828

Reviewed by Ryosuke Niwa.

Generate an unexpected_results.json when comparing results by run.
Update logging to use clearer wording and less unnecessary output.

* Tools/Scripts/compare-static-analysis-results:
(compare_project_results_to_expectations):
(compare_project_results_by_run):
(main):

Canonical link: https://commits.webkit.org/281096@main
